File: LogunovDaniil/Term2nd/Block3/BlackjackSimulatorModified/DeckLibrary/Card.cs
namespace DeckLibrary
{
	public class Card
	{
		public CardSuits Suit { private set; get; }
		public CardRanks Rank { private set; get; }

		public Card(CardSuits suit, CardRanks rank)
		{
			this.Suit = suit;
			this.Rank = rank;
		}

		public override string ToString()
		{
			char[] str = { (char)Suit, (char)Rank };
			return new string(str);
		}
	}
}

File: IvanShurenkov/Second Term/Block2/Task1/lib/base/src/main/kotlin/lib/blackjack/base/Card.kt
package lib.blackjack.base

import lib.blackjack.base.state.CardRank
import lib.blackjack.base.state.CardSuit

class Card(_rank: CardRank = CardRank.QUEEN, _suit: CardSuit = CardSuit.HEARTS) {
    val rank: CardRank
    val suit: CardSuit

    var count: Int
        private set
        get() {
            if (field in 2..10)
                return field
            return 11
        }

    init {
        count = _rank.toInt()
        rank = _rank
        suit = _suit
    }

    override fun toString(): String {
        return rank.toString() + suit.toString()
    }
}